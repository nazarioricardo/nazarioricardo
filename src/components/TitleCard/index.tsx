import styles from "./styles.module.css";

type TitleCardProps = {
  children: React.ReactNode;
  className?: string;
};

function TitleCard({ children, className }: TitleCardProps) {
  return (
    <section
      className={`${styles.titleCard}${className ? " " + className : ""}`}
    >
      {children}
    </section>
  );
}

export default TitleCard;
