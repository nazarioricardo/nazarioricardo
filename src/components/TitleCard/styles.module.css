.titleCard {
  background-color: rgb(var(--surface-rgb));
  padding: var(--padding-vertical) var(--padding-horizontal);
  background: linear-gradient(rgba(0, 117, 124, 1), rgba(0, 79, 123, 1));
  transition: 0.5s;
  border-top-right-radius: 100px;
  border-bottom-right-radius: 100px;
  border-bottom-left-radius: 100px;
  border-top-left-radius: 100px;
}

@media only screen and (max-width: 1170px) {
  .titleCard {
    min-width: 100%;
    max-width: 100%;
    padding: 32px;
    min-height: auto;
    height: auto;
    border-radius: 50px;
    z-index: 0;
  }
}
