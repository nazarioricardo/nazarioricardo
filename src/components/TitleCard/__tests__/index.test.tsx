import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import TitleCard from "../index";

describe("TitleCard", () => {
  test("renders TitleCard component", () => {
    render(<TitleCard className="test-class">Test Content</TitleCard>);
    const sectionElement = screen.getByText("Test Content");
    expect(sectionElement).toBeInTheDocument();
    expect(sectionElement).toHaveClass("test-class");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <TitleCard className="test-class">Test Content</TitleCard>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
