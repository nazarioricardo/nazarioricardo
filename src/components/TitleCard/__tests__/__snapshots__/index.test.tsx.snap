// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleCard matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="titleCard test-class"
  >
    Test Content
  </section>
</DocumentFragment>
`;
